### DISCUSSION TITLE
Feature Request: Open Page in Editor

### DISCUSSION THREAD

**Discussion by jaredpalmer** [vercel/next.js#19716] [Ideas]
# Feature request

As a user, I want to be able to add an `open in editor` link during development so that I can quickly jump to edit content locally (not just when there are errors). 

## Is your feature request related to a problem? Please describe.

Writing docs, being able to jump back to the editor would be huge. This is already supported for errors, but why not all the time.

## Describe the solution you'd like

- An endpoint like `/_next/open-in-editor`, this could then be used to construct a url with the router like `<a href="/_next/open-in-editor?${router.route}.mdx`">Open in Editor</a>`

## Describe alternatives you've considered

- Could be useful in prod too
- Abstract this away into a function or component etc.

**Comment by thomvaill** [comment]
In the mean time, I suggest you create an API route in your project and directly use [launch-editor](https://www.npmjs.com/package/launch-editor), which is the package used by Next.js to open a file in your current editor.

I've used this solution in my [project](https://github.com/thomvaill/log4brains/), if you want to see an example:

- API route `POST /adr/:slug/_open-in-editor`: https://github.com/thomvaill/log4brains/blob/master/packages/web/src/pages/api/adr/%5B...slugAndMore%5D.ts#L17
- launch-editor code: https://github.com/thomvaill/log4brains/blob/master/packages/core/src/infrastructure/api/Log4brains.ts#L168

Instead of using a `slug` param (this is my resource ID), you could just pass your file path.
Of course, this example is from a complex project, you should simplify a lot (should take only a few lines).
